src/server/api/routers/smart-home/entity-state.ts

```diff
@@ -0,0 +1,51 @@
+import { ZodError, z } from 'zod';
+import { HomeAssistantSingleton } from '~/tools/singleton/HomeAssistantSingleton';
+
+import { TRPCError } from '@trpc/server';
+import Consola from 'consola';
+import { findAppProperty } from '~/tools/client/app-properties';
+import { getConfig } from '~/tools/config/getConfig';
+import { createTRPCRouter, protectedProcedure } from '../../trpc';
+
+export const smartHomeEntityStateRouter = createTRPCRouter({
+  retrieveStatus: protectedProcedure
+    .input(
+      z.object({
+        configName: z.string(),
+        entityId: z.string().regex(/^[A-z0-9-_\.]+$/)
```

Comment:
Okay I just understood how this works. We should actually check for `/^[A-Za-z0-9\-_\.]+$/` instead, because otherwise the characters between will also be included:
![image](https://github.com/ajnart/homarr/assets/63781622/6ad4208b-b16e-44a4-957d-e81bdc0a0bc1)